### ISSUE TITLE
🐛 Bug Report: External links in the Shortcuts plugin are linked as internal

### ISSUE THREAD

**Issue by Vity01** [backstage/backstage#18173]
### 📜 Description

External links don't work in ShortCuts plugin. The links are prefixed with a host name (which works for internal links, not for external ones)

### 👍 Expected behavior

links should be opened in the new browser window

### 👎 Actual Behavior with Screenshots

![image](https://github.com/backstage/backstage/assets/2512959/00b2194c-af45-42a7-943a-814ce7266941)

### 👟 Reproduction steps

![image](https://github.com/backstage/backstage/assets/2512959/6e2d3af6-3db5-4195-9735-c0fb2cd94823)
Enable external links support in Shortcuts
`<Shortcuts allowExternalLinks={true}/>`
add any external link, like `https://google.com`
Wrong link is generated then.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by Rugvip** [comment]
Looks like the external link you provided might have an error? Looks like it's `https:/google.com` instead of `https://google.com`?

Regarding opening in an external window, it's intended behavior to not have it force an external window and only do that if you tell the browser to open in a new window (e.g. middle-click)

**Comment by Vity01** [comment]
> Looks like the external link you provided might have an error? Looks like it's `https:/google.com` instead of `https://google.com`?

That might be the bug. I inserted the link correctly in the dialog.

> Regarding opening in an external window, it's intended behavior to not have it force an external window and only do that if you tell the browser to open in a new window (e.g. middle-click)
OK, even most of users would expect such behavior...

**Comment by Vity01** [comment]
I made a short video to demo bug.
![bug](https://github.com/backstage/backstage/assets/2512959/d043db54-a07d-4a33-bff5-73ef8625a4d3)

**Comment by Rugvip** [comment]
@Vity01 I'm not able to reproduce this issue, when I add a link to `https://google.com` it works as expected. What version of Backstage are you on?

**Comment by Vity01** [comment]
```
{
  "version": "1.14.1"
}
```

**Comment by freben** [comment]
I've tried it out now as well, and can't make it fail. I add the exact same link as you, and the shortcut turns out perfectly right. Hm.

**Comment by freben** [comment]
Two more thought.

The shortcuts API hooks itself up to the storageApi. By default that one stores data in the browser's local storage. Did you replace it with the user-settings-backend powered database store instead? Shouldn't make a difference, but if you did, you would be able to look at the network traffic going back and forth as it was saved.

If you didn't, can you just double check what your local storage version of the data looks like? In the Application tab of your browser dev tools.

![Screenshot 2023-06-16 at 10 47 59](https://github.com/backstage/backstage/assets/3097461/4c2e01a9-c7c6-4312-ac6d-4764fe75dfae)

**Comment by Vity01** [comment]
Screenshot from my LocalStorage (I also used Private mode to be sure all previous cookies are not present)

![image](https://github.com/backstage/backstage/assets/2512959/83eb29d6-17ee-48fe-9a25-5f5112634012)
It seems to be correct, but the clickable link is still incorrect.

**Comment by Vity01** [comment]
I made a small discovery. This is how the generated source code looks like 
![image](https://github.com/backstage/backstage/assets/2512959/1f01b201-945e-4e6f-8140-4e3510cdc4d9)

if I modify it into correct `https://google.com` the result target link is also correct.

**Comment by Vity01** [comment]
Reporting the problem persist on v1.15 as well.

**Comment by Rugvip** [comment]
Still not able to reproduce this O_o. Tried reloading the page too.

Do you happen to have a custom or otherwise non-standard `StorageApi` installed?

**Comment by Vity01** [comment]
I am not sure what you mean by non-standard StorageAPI. I don't think we use anything special. 
I also put a breakpoint into ShortCutsAPI and data are loaded properly.
![image](https://github.com/backstage/backstage/assets/2512959/5d8bedfe-1fc3-4713-b4cc-c78d90d6bd84)

The problem is somewhere on the presentation layer - in the SiderBartItem, Theme or something similar.

**Comment by Vity01** [comment]
I found it!
I started to compare dependencies with current BS `master` and I found older versions under `packages/app/package.json`
![image](https://github.com/backstage/backstage/assets/2512959/f9ed2966-263a-406f-ae62-4adfaba9264f)
Upgrading React helped to fix it. 
I don't understand why auto-upgrading does not change these versions too. I am going to compare and sync our "base" with current base. I found other many changes under root `package.json`.
